tests/src/tests/mod.rs

```diff
@@ -351,3 +365,44 @@ async fn test_successful_complicated_flow() {
     assert_eq!(usdc.ft_balance_of(silo2.id()).await, 5_700_000);
     assert_eq!(usdc.ft_balance_of(silo3.id()).await, 7_600_000);
 }
+
+#[tokio::test]
+async fn test_storage_deposit_refund() {
+    use crate::sandbox::factory::Factory;
+
+    let sandbox = Sandbox::new().await.unwrap();
+    let fees = sandbox.deploy_fees(&[]).await.unwrap();
+    let (wnear, _) = sandbox.deploy_wrap_near().await.unwrap();
+    let factory = sandbox.deploy_factory(fees.id()).await.unwrap();
+    let balance_before_create = sandbox.balance(factory.id()).await;
+    let parameters = DeployParameters {
+        target_address: Address::from_array([1; 20]).encode(),
+        target_network: "silo.test.near".parse().unwrap(),
+        wnear_contract_id: wnear.id().as_str().parse().unwrap(),
+    };
+
+    let forwarder_ids = factory.create(&[parameters]).await.unwrap();
+    assert_eq!(forwarder_ids.len(), 1);
+
+    let balance_after_create = sandbox.balance(factory.id()).await;
+    println!("balance_after_create: {balance_after_create}");
+    assert_eq!(
+        balance_before_create - balance_after_create,
+        363_685_846_049_881_000_000_000 // 0.363685846049881 Ⓝ
```

Comment:
It's hard to exactly account for $NEAR spent on gas. It might make sense to change this assert to only check the first four significant digits or something.